### PR TITLE
Add the option to move stories to another project

### LINKED ISSUE #132: Add an option to move a story
**Branch/Commit:**

Main branch

**Describe the feature:**

It would be useful to have an option to move a story from a sub-project to a sibling sub-project, keeping the estimates too.

**Problem:**

Sometimes a story should belong to a different sub-project, currently we have to delete the original story and create a new one. This is not only time consuming but also clears the estimates if those where already done.

**I will abide by the [code of conduct](https://github.com/fastruby/points/blob/main/CODE_OF_CONDUCT.md)**

### PR DESCRIPTION
**Pull request by arielj**
**Description:**

This PR adds the new feature of moving stories between projects.

- a new `Move to` button is added for each row if the current project is a subproject with siblings
- clicking `Move to` shows all possible siblings of the current project
- after confirmation, the story is moved and the user is redirected back to the initial project

I added a `within_story_row` to abstract the `within("#story_#{story.id}")` selector and updated that in some places

**How to QA/test:**

- Create a project
- Create a few subprojects
- Create a story on any of the subprojects
- The row of the story should have a `Move to` button
- Click the `Move to` button and select another project and confirm the move

The `Move to` button only appears if a project is a subproject and if it has siblings, we don't want to move stories between any possible project.

Closes #132

___

I will abide by the [code of conduct](https://github.com/fastruby/points/blob/main/pull_request_template.md).

### CODE REVIEW

File: spec/controllers/stories_controller_spec.rb [hunk 1]
```diff
@@ -107,4 +107,16 @@
...
   end
+
+  describe "#move" do
+    it "does not allow moving stories to non-sibling projects" do
+      project2 = FactoryBot.create(:project, parent: project)
+      project3 = FactoryBot.create(:project)
+      story = FactoryBot.create(:story, project: project2)
+
+      put :move, params: {project_id: project2.id, story_id: story.id, to_project: project3.id}
+      expect(flash[:error]).to eq "Selected project does not exists or is not a sibling."
```
Comment (lubc) [inline spec/controllers/stories_controller_spec.rb:118]:
```suggestion
      expect(flash[:error]).to eq "Selected project does not exist or is not a sibling."
```

File: app/controllers/stories_controller.rb [hunk 2]
```diff
@@ -96,14 +96,27 @@ def render_markdown
...
   end
 
+  def move
+    @new_project = @project.siblings.find_by(id: params[:to_project])
+
+    if @new_project
+      @story.update_attribute(:project_id, @new_project.id)
+      flash[:success] = "Story moved!"
+    else
+      flash[:error] = "Selected project does not exists or is not a sibling."
```
Comment (lubc) [inline app/controllers/stories_controller.rb:106]:
```suggestion
      flash[:error] = "Selected project does not exist or is not a sibling."
```

**Review by lubc — APPROVED** [review]
Looks good! 

Just commented one small typo in the error message

**Comment by lubc** [comment]
@neutrino Could you please QA this PR? Thanks!

**Comment by arielj** [comment]
@lubc I fixed the typo and the merge conflicts :+1:

**Review by ranendra-modus — APPROVED** [review]
LGTM

**Comment by arielj** [comment]
Tests in the next version seems to be broken by a change in Rails, I fixed that in this PR https://github.com/fastruby/points/pull/147

**Comment by cleicar** [comment]
@arielj now that #147 is merge I believe we just need to update this branch, right?

**Comment by kindoflew** [comment]
@neutrino, just confirming that this passed QA (because the 'Ready to QA' label is still there). thanks!

**Comment by kindoflew** [comment]
@neutrino, just confirming that this passed QA (since the `Ready to QA` label is still there). Thanks!

**Pull request merged**